### ISSUE TITLE
How can we highlight template literals as JavaScript?

### ISSUE THREAD

**Issue by kireerik** [Thom1729/Sublime-JS-Custom#140]
To highlight as `HTML` it works like this:
```JSON
"custom_template_tags": {
	"": "scope:text.html.basic"
}
```

&nbsp;
To highlight as `JavaScript` I tried the following for example:
```JSON
"custom_template_tags": {
	"": "scope:source.js"
}
```

**Comment by Thom1729** [comment]
It's not clear to me what you're asking. You might be looking for the [`default`](https://github.com/Thom1729/Sublime-JS-Custom/blob/master/docs/configuration.md#user-content-default-string) configuration option.

**Comment by kireerik** [comment]
It works like that. Thank you!

```JSON
"custom_templates": {
	"default": "scope:source.js"
}
```

**Comment by kireerik** [comment]
How can I restore the default highlighting for template strings (in `lookaheads`)?

I know how to set `lookaheads`, but I don't know what should the value be to reset to the default (in a `js` file).

**Comment by Thom1729** [comment]
You want e.g. templates to be highlighted as HTML by default, but as plain template strings with a lookahead? I'm having trouble envisioning the use case; can you provide an example?

If that's what you mean, then I'm not sure there's a clean solution at the moment. You could do it as follows:

```json
"custom_templates": {
  "default": "scope:text.html.basic",
  "lookaheads": {
    "foo": "string-content"
  }
},
```

But a) you won't get the proper string scope, and b) this is kind of hacky and brittle. If this is a real use case for you, then it might not be too much of a code change to support it.

**Comment by Thom1729** [comment]
Rather than `".": "string-content"`, you need `"\\.": "string-content"`. The key there is interpreted as a regexp, so the dot will match anything; you need a backslash to escape the dot, and then another backslash for JSON escaping. Without that, any string will match the lookahead, so you'll never see the default.

This seems to work for me, with the caveat that (as mentioned above) e.g. `./foo` won't get the normal string scope (and color).

Alternatively, what about:

```json
"custom_templates": {
  "lookaheads": {
    "\\<": "scope:text.html.basic",
    "import": "scope:source.js",
  },
},
```

**Comment by kireerik** [comment]
The alternative is not working.

**Comment by kireerik** [comment]
Seems like the solution is to not define the `default` at all. Like this for example:
```JSON
"custom_templates": {
	"lookaheads": {
		"\\<|.\\>": "scope:text.html.basic"

		, "var": "scope:source.js"
	}
}
```

**Comment by kireerik** [comment]
Yes, it works like this for me. Thank you!

**Comment by kireerik** [comment]
Similarly to template strings can I set (custom) highlighting (using lookaheads) for regular strings?

**Comment by Thom1729** [comment]
That is not currently a feature. Given that you can use template strings anywhere, I suspect that custom embedding for single/double-quoted strings would not be useful enough to justify the complexity. In addition, there might be substantial performance implications. What is your use case for this?

**Comment by kireerik** [comment]
Yes, it would be just nice to have. I have some single line html strings and `'`  looks simpler and it is easier to type in these cases compared to `.

**Comment by kireerik** [comment]
I am implementing JavaScript Extensible Markup Language usage for some parts that are currently modules and will become (SolidJS) components in the [Refo](https://github.com/kireerik/refo) static site example.

So for example the following line:
- [index/index/site/main/index.js#L20](https://github.com/kireerik/refo/blob/48072d16f27d007e5c16d628cec1c00a91a67235/example/index/index/site/main/index.js#L20)

will look like this:
```JavaScript
{template(`<!DOCTYPE HTML>`)}
```

and unfortunately this line too for example:
- [#L22](https://github.com/kireerik/refo/blob/48072d16f27d007e5c16d628cec1c00a91a67235/example/index/index/site/main/index.js#L22)

```JavaScript
{template(`<head>`)}
```
